### PR TITLE
Upgrade to TypeScript 2.5

### PR DESCRIPTION
**Pull request by alexeagle**
- update to TypeScript 2.5
- point the 2.4 typings test at the previous typescript version, so we
  don't break it accidentally
- update to latest TypeScript 2.5 compatible Bazel rules
- move .bazelrc to tools/bazel.rc per https://docs.bazel.build/versions/master/best-practices.html#bazelrc

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

### CODE REVIEW

**Review by IgorMinar — APPROVED** [review]
Please fix the commit message format. otherwise lgtm supposing that ci passes.

**Comment by alexeagle** [comment]
@chuckjaz I hear that we need a single tsickle version that works with all supported typescript versions. Tobias says you have an idea for that.

File: packages/compiler-cli/src/transformers/compiler_host.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ export class TsCompilerAotCompilerTypeCheckHostAdapter implements ts.CompilerHos
         return sf ? this.metadataProvider.getMetadata(sf) : undefined;
       },
       fileExists: (filePath) => this.originalFileExists(filePath),
-      readFile: (filePath) => this.context.readFile(filePath),
+      readFile: (filePath) => this.context.readFile(filePath) !,
```
Comment (chuckjaz) [inline packages/compiler-cli/src/transformers/compiler_host.ts]: Can you add a clean-up TODO comment here and other places that are similar below?

**Review by chuckjaz — APPROVED** [review]
One nit and a request to add a typing_test_ts25.

File: packages/compiler-cli/src/transformers/compiler_host.ts (hunk 1 above)
Comment (alexeagle) [reply packages/compiler-cli/src/transformers/compiler_host.ts]: Sure, added an `assert` helper at the top of the file with a TODO for you. I don't want to change the behavior to throw in this PR.

**Review by mhevery — APPROVED** [review]

**Comment by alexeagle** [comment]
Note, if you call `Injector.get(Token)` with a Token with static members, you'll get the wrong type, see https://github.com/Microsoft/TypeScript/issues/20102

**Pull request closed without merging**